pkg/util/containers/metrics/cgroup_metrics.go

```diff
@@ -296,18 +302,37 @@ func (c ContainerCgroup) IO() (*CgroupIOStat, error) {
 	}
 	defer f.Close()
 
+	// Get device id->name mapping
+	var devices map[string]string
+	mapping, err := getDiskDeviceMapping()
+	if err != nil {
+		log.Debugf("Cannot get per-device stats: %s", err)
+	} else {
+		devices = mapping.idToName
+	}
+
 	scanner := bufio.NewScanner(f)
 	for scanner.Scan() {
 		fields := strings.Split(scanner.Text(), " ")
+		if len(fields) < 3 {
+			continue
+		}
+		deviceName := devices[fields[0]]
```

Comment:
`getDiskDeviceMapping` is able to return `nil` so this line could potentially crash?

Reply:
Lookups in `nil` maps are actually safe: https://play.golang.org/p/Ms51BMTQocP
but that might indeed be playing with fire